13.0/s2i/bin/assemble

```diff
@@ -171,7 +172,43 @@ function configure_proxy() {
   xml="$xml\
        </proxy>"
     sed -i "s|<!-- ### configured http proxy ### -->|$xml|" $HOME/.m2/settings.xml
+    return
   fi
+
+  # if http(s)_proxy is set, use that.
+  proxy=${HTTPS_PROXY:-${https_proxy:=${HTTP_PROXY:=$http_proxy}}}
+  noproxy=${NO_PROXY:-$no_proxy}
+  if [ -n $proxy ]; then
+    proxyhost=$(echo $proxy | cut -d : -f 1,2)
+    proxyport=$(echo $proxy | cut -d : -f 3)
+    proxyprotocol="http"
+    if [[ `echo $proxy | grep -i https://` ]]; then
+      proxyport=${proxyport:-443}
+      proxyprotocol="https"
+    else
+      proxyport=${proxyport:-80}
+    fi
+
+    xml="<proxy>\
+         <id>genproxy</id>\
+         <active>true</active>\
+         <protocol>$proxprotocol</protocol>\
+         <host>$proxyhost</host>\
+         <port>$proxyport</port>"
+    if [ -n "$HTTP_PROXY_USERNAME" -a -n "$HTTP_PROXY_PASSWORD" ]; then
```

Comment:
I'd like to see us support the PROXY_USERNAME/PROXY_PASSWORD going forward and have some code here to support legacy configurations until we can deprecate it?